src/library_fs.js

```diff
@@ -427,14 +427,18 @@ FS.staticInit();` +
       // clone it, so we can return an instance of FSStream
       stream = Object.assign(new FS.FSStream(), stream);
       if (fd == -1) {
-        fd = FS.nextfd();
+        fd = FS.streams.allocate(stream);
+      } else {
+        while (FS.streams.allocated.length <= fd) {
+          FS.streams.allocated.push();
+        }
```

Comment:
Should this be a helper function on HandleAllocator? Or is it not a repeating pattern?